index.js

```diff
@@ -6,6 +6,7 @@ const {execFile} = require('child_process');
 const execFileP = util.promisify(execFile);
 const bin = path.join(__dirname, 'file-icon');
 const HUNDRED_MEGABYTES = 1024 * 1024 * 100;
+const EOF = '<EOF>';
```

Comment:
What if an image contains this string? For example, as metadata or something else. Should we try to use something entirely unique

Reply:
Thinking about the actual issue better, I don't think that outputting multiple buffers to stdout is the best idea.

I realize that the description of the issue specifically said that the swift executable should only be invoked once / output multiple buffers at once, but I believe that's a bit misguided. I don't see an issue in making multiple calls to the swift executable from JS, instead of one and then splitting buffers with an arbitrary separator.

The changes required on both sides to accomplish this made the project really grow in complexity for no real gain. I was originally going to submit a "multiple-calls-to-swift" implementation, but then realized that a very similar PR was rejected because it didn't follow the issue's description.

Thoughts?

Reply:
The motivating factor of requiring only one call to the Swift binary was that spawning a binary in Node.js is slow and I was optimizing for the case where Swift handles the writing too (`fileIcon.file()`), as in that case, it would be wasteful to spawn the binary many times.

How about `fileIcon.file()` continues to use a single call, but `fileIcon.buffer()` uses multiple calls? For the `fileIcon.buffer()` case, we could even make it concurrent with a sane limit like 8. 

https://github.com/sindresorhus/p-map

Reply:
That is indeed a good approach that would remove the need for hacks. I'll try and have the changes up soon

Reply:
@sindresorhus Updated with `p-map` and concurrent calls to the swift binary in `buffer` mode